### ISSUE TITLE
[debops.users] Failing at home creation

### ISSUE THREAD

**Issue by thiras** [debops/debops#1456]
Here is my users.yml file;

```yaml
users__host_accounts:
  - name: 'example'
    group: 'example'
    shell: '/bin/bash'
    home: '/srv/www/example'
    generate_ssh_key: True

  - name: 'example2'
    group: 'example2'
    shell: '/bin/bash'
    home: '/srv/www/example2'
    generate_ssh_key: True
```

Here is the error I've got;

```
TASK [users : Manage user accounts] *********************************************************************
[WARNING]: 'local: true' specified and user 'example' was not found in /etc/passwd. The local user
account may already exist if the local account database exists somewhere other than /etc/passwd.
failed: [production0] (item={'name': 'example', 'state': 'present', 'gecos': ''}) => changed=false 
  ansible_loop_var: item
  item:
    generate_ssh_key: true
    group: example
    home: /srv/www/example
    id: 0
    name: example
    real_weight: 0
    separator: false
    shell: /bin/bash
    state: present
    weight: 0
  msg: |-
    Error creating /srv/www/example: Error creating `/srv/www/example': File exists.
  name: example
  rc: 7
changed: [production0] => (item={'name': 'example2', 'state': 'present', 'gecos': ''})
[WARNING]: 'local: true' specified and user 'example2' was not found in /etc/passwd. The local user
account may already exist if the local account database exists somewhere other than /etc/passwd.
```
I've tried both with `create_home: True` and `create_home: False`. Both didn't help.

However, the script works fine at the re-run.

**Comment by drybjed** [comment]
Does the `/srv/www/example` directory exist before this task is executed? Try removing it or changing the home directory, see if the error shows up again.

**Comment by thiras** [comment]
@drybjed No. It's a fresh machine.

To reproduce;

1. `debops bootstrap -u root`
2. `debops`

Just guessing but maybe the problem is bootstrapping since it creates the users

**Comment by drybjed** [comment]
The problem seems to be confined to the `users` role which is part of the `bootstrap` playbook. And you say this is reproductible?

As for debugging it... Run the `bootstrap` playbook and when the task fails, login to the host either as your user or as `root` and look around. Check if the account is present in `getent passwd`, check if the `/srv/www/example` directory is present. Also, is `/srv/www` directory present? because it's not created by default; in that case that would explain the error. What happens if you change the home directory to `/home/example`?

**Comment by tasosalvas** [comment]
I had a similar issue in an older iteration of my own thing and I did have to create `/srv/www` manually before users ran. I didn't comment earlier because the error states the exact opposite... It does explain why it works after nginx has had a chance to create the dir.

My current app users live at `/home` and I have no such issue... maybe `debops.resources` runs early enough to do the job?

**Comment by drybjed** [comment]
The `resources` role is not included in the `bootstrap` playbook, but another role is: `fhs`. This role is meant for that specific task - cration of base filesystem directories. By default it does not create the `/srv/www` directory, since not all hosts are webservers (I could add that as default if enough people want to, though). But you can easily add it by including in the Ansible inventory:

```yaml
fhs__directories:
  - name: 'www'
    path: '/srv/www'
```
This will also create the `ansible_local.fhs.www` Ansible local fact which can be used elsewhere to point to that base directory.

**Comment by tasosalvas** [comment]
One day I will know my way around the project I swear :).

I guess it could sneakily check inventory vars to see if nginx is going to be using it later, but it'd be a bit of a kludge and wouldn't address other arbitrary basedirs. `fhs` sounds like the proper solution.

I'll add a warning and a link to `fhs` in `users > defaults_detailed > item.home`, I think it's all that's needed.

**Comment by thiras** [comment]
> The problem seems to be confined to the `users` role which is part of the `bootstrap` playbook. And you say this is reproductible?
> 
> As for debugging it... Run the `bootstrap` playbook and when the task fails, login to the host either as your user or as `root` and look around. Check if the account is present in `getent passwd`, check if the `/srv/www/example` directory is present. Also, is `/srv/www` directory present? because it's not created by default; in that case that would explain the error. What happens if you change the home directory to `/home/example`?

Yes, it's reproducible. The error occurs at running `debops` after running `debops bootstrap`. I'll check if the directories created after bootstrap and let you know.

**Comment by thiras** [comment]
I've followed the 30205c8 warning. Now everything seems OK.
